packages/flutter_test/lib/src/widget_tester.dart

```diff
@@ -604,34 +603,11 @@ class WidgetTester extends WidgetController implements HitTestDispatcher, Ticker
     }
   }
 
-  /// Repeatedly calls [pump] with the given `duration` until there are no
-  /// longer any frames scheduled. This will call [pump] at least once, even if
-  /// no frames are scheduled when the function is called, to flush any pending
-  /// microtasks which may themselves schedule a frame.
-  ///
-  /// This essentially waits for all animations to have completed.
-  ///
-  /// If it takes longer that the given `timeout` to settle, then the test will
-  /// fail (this method will throw an exception). In particular, this means that
-  /// if there is an infinite animation in progress (for example, if there is an
-  /// indeterminate progress indicator spinning), this method will throw.
-  ///
-  /// The default timeout is ten minutes, which is longer than most reasonable
-  /// finite animations would last.
-  ///
-  /// If the function returns, it returns the number of pumps that it performed.
-  ///
-  /// In general, it is better practice to figure out exactly why each frame is
-  /// needed, and then to [pump] exactly as many frames as necessary. This will
-  /// help catch regressions where, for instance, an animation is being started
-  /// one frame later than it should.
-  ///
-  /// Alternatively, one can check that the return value from this function
-  /// matches the expected number of pumps.
+  @override
   Future<int> pumpAndSettle([
```

Comment:
If we're going to do this breaking change, I felt a named parameter map `[duration], {timeout, phase}` is better than a positional one such as `[duration, timeout, phase]`. It should be compatible with most `pumpAndSettle` calls with just 0 or 1 positional argument, and will force `timeout, phase` to be spelled out if used.

Reply:
if we're changing the API anyway, then definitely just drop timeout entirely.

Reply:
Removing timeout is a real breaking change, tested here: https://github.com/flutter/flutter/blob/67e9b2945ec6b09d4c1719a9aa81578e95cd6071/packages/flutter_test/test/widget_tester_test.dart#L473-L478 @Hixie